src/agentconfig.rs

```diff
@@ -228,6 +289,51 @@ impl Configurable for AgentConfig {
         };
         info!("Selected {} as local address to listen on.", final_ip);
 
+        // Parse data directory from values
+        let final_data_dir = if let Ok(data_dir) =
+            AgentConfig::get_exactly_one_string(&parsed_values, &AgentConfig::DATA_DIR)
+        {
+            PathBuf::from_str(&data_dir).unwrap_or_else(|_| {
+                panic!(
+                    "Error parsing valid data directory from string: {}",
+                    data_dir
+                )
+            })
+        } else {
+            // Should not happen due to default value being assigned to the parameter
+            PathBuf::from("")
+        };
+
+        // Parse bootstrap file from values
+        let final_bootstrap_file = if let Ok(bootstrap_file) =
+            AgentConfig::get_exactly_one_string(&parsed_values, &AgentConfig::BOOTSTRAP_FILE)
+        {
+            PathBuf::from_str(&bootstrap_file).unwrap_or_else(|_| {
+                panic!(
+                    "Error parsing valid data directory from string: {}",
+                    bootstrap_file
+                )
+            })
+        } else {
+            // Should not happen due to default value being assigned to the parameter
+            PathBuf::from("")
```

Comment:
If this should not happen wouldn't it make more sense to cause a panic if it does? Especially as this is on startup it'd catch programming errors.

So basically change the if let to something like ...`unwrap()`
I haven't fully thought it through but should work, no?

Reply:
I've extracted this code into a function and refactored it a bit.